src/tree/values/FormattingElement.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Represents formatting elements (e.g. <strong>, <a>, <em>), except text, within an HTML-text.
+ */
+class FormattingElement {
+	/**
+	 * Represents a formatting element (e.g. <strong>, <a>, <em>), except text, within an HTML-text.
+	 *
+	 * @param {string} tag          		The HTML-tag of this element (e.g. "strong", "a", etc.).
+	 * @param {number} start        		The start position of the tag within the text.
+	 * @param {number} end               	The end position of the tag within the text.
+	 * @param {Object} [attributes=null] 	The attributes (as key-value pairs, e.g. "href='...'" => { href: '...' } ).
+	 * @param {boolean} [selfClosing=false]	If this (HTML) element is self-closing (like for `img` elements).
+	 */
+	constructor( tag, start, end, attributes = null, selfClosing = false ) {
+		this.tag = tag;
+		this.attributes = attributes;
+		this.start = start;
+		this.end = end;
+		this.selfClosing = selfClosing;
+		this._validate();
+	}
+
+	/**
+	 * Checks if this is a valid FormattingElement. If it is not, throws an error.
+	 *
+	 * @private
```

Comment:
This private reminds me, there is also a `@throws` you can use for the `RangeError` http://usejsdoc.org/tags-throws.html 

Also, the upper border of the end could be checked for too right?
Or we could just reset `start < 0` to 0 and `> end` to end. And/or swap the start/end with a warning instead.
Possibilities 💥

Reply:
The code now tries to fix the mistakes and gives a warning in the console when it does.

This does have the possibility to mess the formatting up. For example formatting elements can overlap after "fixing" it. However, trying to solve this would lead to into a rabbit hole that I do not want to get into... Therefor, I would suggest leaving it at this.